### ISSUE TITLE
Error with Hammerspoon Version 0.9.75 (0.9.75)

### ISSUE THREAD

**Issue by cnshsliu** [agzam/spacehammer#14]
... shows some errors in the console ...

**Comment by agzam** [comment]
The project is being re-written in Fennel. https://github.com/agzam/spacehammer#fennel Have you installed all pre-requisites?

**Comment by erikcw** [comment]
@agzam I'm experiencing the exact same issue.  Followed the README to the letter (I think!).  I think it has something to do with the Fennel package loader not being used (defined in init.lua).

```
2019-12-16 16:25:48: -- Loading extension: ipc
2019-12-16 16:25:48: -- Loading extension: hints
2019-12-16 16:25:48: -- Loading extension: uielement
2019-12-16 16:25:48: -- Loading extension: window
2019-12-16 16:25:48: -- Loading extension: pathwatcher
2019-12-16 16:25:48: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: module 'modal' not found:
	no field package.preload['modal']
	no file '/Users/erik/.hammerspoon/modal.lua'
	no file '/Users/erik/.hammerspoon/modal/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/modal.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/modal.lua'
	no file '/usr/local/share/lua/5.3/modal/init.lua'
	no file '/usr/local/lib/lua/5.3/modal.lua'
	no file '/usr/local/lib/lua/5.3/modal/init.lua'
	no file './modal.lua'
	no file './modal/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal/init.lua'
	no file '/Users/erik/.hammerspoon/modal.so'
	no file '/usr/local/lib/lua/5.3/modal.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './modal.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/core.lua:30: in main chunk
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/init.lua:10: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
2019-12-16 16:26:11: -- Loading extension: fnutils
2019-12-16 16:26:11: -- Lazy extension loading enabled
2019-12-16 16:26:11: -- Loading ~/.hammerspoon/init.lua
2019-12-16 16:26:11: -- Loading extension: alert
2019-12-16 16:26:11: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: module 'core' not found:
	no field package.preload['core']
	no file '/Users/erik/.hammerspoon/core.lua'
	no file '/Users/erik/.hammerspoon/core/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/core.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/core.lua'
	no file '/usr/local/share/lua/5.3/core/init.lua'
	no file '/usr/local/lib/lua/5.3/core.lua'
	no file '/usr/local/lib/lua/5.3/core/init.lua'
	no file './core.lua'
	no file './core/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core/init.lua'
	no file '/Users/erik/.hammerspoon/core.so'
	no file '/usr/local/lib/lua/5.3/core.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './core.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/init.lua:10: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
2019-12-16 16:26:21: -- Lazy extension loading enabled
2019-12-16 16:26:21: -- Loading ~/.hammerspoon/init.lua
2019-12-16 16:26:21: -- Loading extension: alert
2019-12-16 16:26:21: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: module 'core' not found:
	no field package.preload['core']
	no file '/Users/erik/.hammerspoon/core.lua'
	no file '/Users/erik/.hammerspoon/core/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/core.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/core.lua'
	no file '/usr/local/share/lua/5.3/core/init.lua'
	no file '/usr/local/lib/lua/5.3/core.lua'
	no file '/usr/local/lib/lua/5.3/core/init.lua'
	no file './core.lua'
	no file './core/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core/init.lua'
	no file '/Users/erik/.hammerspoon/core.so'
	no file '/usr/local/lib/lua/5.3/core.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './core.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/init.lua:10: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
```

If I manually compile `core.fnl` (`fennel --compile core.fnl > core.lua`) then it moves on to not being able to load model.lua, and so on.

```
2019-12-16 16:27:19: -- Loading extension: ipc
2019-12-16 16:27:19: -- Loading extension: hints
2019-12-16 16:27:19: -- Loading extension: uielement
2019-12-16 16:27:19: -- Loading extension: window
2019-12-16 16:27:19: -- Loading extension: pathwatcher
2019-12-16 16:27:19: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: module 'modal' not found:
	no field package.preload['modal']
	no file '/Users/erik/.hammerspoon/modal.lua'
	no file '/Users/erik/.hammerspoon/modal/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/modal.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/modal.lua'
	no file '/usr/local/share/lua/5.3/modal/init.lua'
	no file '/usr/local/lib/lua/5.3/modal.lua'
	no file '/usr/local/lib/lua/5.3/modal/init.lua'
	no file './modal.lua'
	no file './modal/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal/init.lua'
	no file '/Users/erik/.hammerspoon/modal.so'
	no file '/usr/local/lib/lua/5.3/modal.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './modal.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/modal.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/core.lua:30: in main chunk
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/init.lua:10: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
```
`fennel --compile modal.fnl > model.lua`
```
2019-12-16 16:28:06: -- Loading extension: ipc
2019-12-16 16:28:06: -- Loading extension: hints
2019-12-16 16:28:06: -- Loading extension: uielement
2019-12-16 16:28:06: -- Loading extension: window
2019-12-16 16:28:06: -- Loading extension: pathwatcher
2019-12-16 16:28:06: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: module 'windows' not found:
	no field package.preload['windows']
	no file '/Users/erik/.hammerspoon/windows.lua'
	no file '/Users/erik/.hammerspoon/windows/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/windows.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/windows.lua'
	no file '/usr/local/share/lua/5.3/windows/init.lua'
	no file '/usr/local/lib/lua/5.3/windows.lua'
	no file '/usr/local/lib/lua/5.3/windows/init.lua'
	no file './windows.lua'
	no file './windows/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/windows.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/windows/init.lua'
	no file '/Users/erik/.hammerspoon/windows.so'
	no file '/usr/local/lib/lua/5.3/windows.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './windows.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/windows.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/modal.lua:4: in main chunk
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/core.lua:30: in main chunk
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/erik/.hammerspoon/init.lua:10: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
```

I'm on Hammerspoon 0.9.76.  Mac OS Catalina.

**Comment by agzam** [comment]
What about `develop` branch? I can't test on Catalina, I haven't upgraded yet, sorry.

**Comment by agzam** [comment]
@erikcw What about full-disk access? I'm guessing here, but have you tried giving Full Disk Access to Hammerspoon in "Settings -> Security & Privacy"?

**Comment by erikcw** [comment]
@agzam I just tested with Full Disk Access and the `develop` branch.  Still the same error.

**Comment by asconix** [comment]
On a fresh installed Catalina (with dependencies installed):

```
2019-12-18 21:43:58: -- Loading extension: ipc
2019-12-18 21:43:58: -- Loading extension: hints
2019-12-18 21:43:58: -- Loading extension: uielement
2019-12-18 21:43:58: -- Loading extension: window
2019-12-18 21:43:58: -- Loading extension: pathwatcher
2019-12-18 21:43:58: *** ERROR: Compile error in 'global' ./modal.fnl:28: global states conflicts with local
stack traceback:
	[C]: in function 'error'
	/usr/local/share/lua/5.3/fennel.lua:473: in upvalue 'assertCompile'
	/usr/local/share/lua/5.3/fennel.lua:1126: in upvalue 'getname'
	/usr/local/share/lua/5.3/fennel.lua:1167: in function </usr/local/share/lua/5.3/fennel.lua:1164>
	(...tail calls...)
	/usr/local/share/lua/5.3/fennel.lua:1493: in local 'special'
	/usr/local/share/lua/5.3/fennel.lua:966: in function '__fennel-bootstrap__.compile1'
	/usr/local/share/lua/5.3/fennel.lua:2117: in function '__fennel-bootstrap__.compileStream'
	(...tail calls...)
	/usr/local/share/lua/5.3/fennel.lua:2243: in function '__fennel-bootstrap__.eval'
	(...tail calls...)
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	./core.fnl:30: in main chunk
	(...tail calls...)
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:579: in function 'require'
	/Users/ctp/.hammerspoon/init.lua:6: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:630: in function 'hs._coresetup.setup'
	(...tail calls...)
```

**Comment by eccentric-j** [comment]
@asconix That's the `master` branch or `develop` branch?

**Comment by agzam** [comment]
Sorry friends, I just tried upgrading Fennel, it seems current master (of Spacehammer) is no longer compatible with the latest implementation of Fennel. `Develop` branch seems to be working fine.

There were major redesigns done by @eccentric-j in `develop` branch, and it's taking for me too long to make it ready to "release".

I think it doesn't make sense trying to fix master (I will try anyway) at this point and it's better to focus on making `develop` ready to be merged.

There are a few things in `master` that I personally still depend on, which I haven't had time to move to develop, but that is not the main reason why I haven't given the "green light". 

In some sense I'm happy that Fennel devs made breaking changes, that would force us to switch to improved version of Spacehammer sooner.

**Comment by agzam** [comment]
fixed in 1.5

**Comment by erikcw** [comment]
@agzam Both `master` and `develop` are giving me the following error:

```
2020-02-04 17:02:32: Welcome to the Hammerspoon Console!
You can run any Lua code in here.

2020-02-04 17:02:32: -- Lazy extension loading enabled
2020-02-04 17:02:33: -- Loading ~/.hammerspoon/init.lua
2020-02-04 17:02:33: -- Loading extension: hotkey
2020-02-04 17:02:33: 17:02:33     hotkey: Enabled hotkey ⌘⌃⌥R
2020-02-04 17:02:33: -- Loading extension: alert
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥W
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥Y
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥K
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥U
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥H
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥L
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥B
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥J
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥N
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥LEFT
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥RIGHT
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥UP
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥DOWN
2020-02-04 17:02:33:              hotkey: Enabled hotkey ✧UP
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥E
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥SPACE
2020-02-04 17:02:33: -- Loading extension: menubar
2020-02-04 17:02:33: -- Loading extension: caffeinate
2020-02-04 17:02:33:              hotkey: Enabled hotkey ⌘⌃⌥C
2020-02-04 17:02:33:              hotkey: Enabled hotkey ✧M
2020-02-04 17:02:33:              hotkey: Enabled hotkey ✧D
2020-02-04 17:02:33: -- Loading Spoon: ControlEscape
2020-02-04 17:02:33: -- Loading extension: timer
2020-02-04 17:02:33: -- Loading extension: eventtap
2020-02-04 17:02:33: -- Done.
2020-02-04 17:04:56: -- Loading extension: window
2020-02-04 17:04:56: -- Loading extension: uielement
2020-02-11 16:38:45: -- Lazy extension loading enabled
2020-02-11 16:38:45: -- Loading ~/.hammerspoon/init.lua
2020-02-11 16:38:45: -- Loading extension: alert
2020-02-11 16:38:45: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:597: module 'core' not found:
	no field package.preload['core']
	no file '/Users/erik/.hammerspoon/core.lua'
	no file '/Users/erik/.hammerspoon/core/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/core.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/core.lua'
	no file '/usr/local/share/lua/5.3/core/init.lua'
	no file '/usr/local/lib/lua/5.3/core.lua'
	no file '/usr/local/lib/lua/5.3/core/init.lua'
	no file './core.lua'
	no file './core/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core/init.lua'
	no file '/Users/erik/.hammerspoon/core.so'
	no file '/usr/local/lib/lua/5.3/core.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './core.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:597: in function 'require'
	/Users/erik/.hammerspoon/init.lua:6: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:648: in function 'hs._coresetup.setup'
	(...tail calls...)
2020-02-11 16:40:10: -- Lazy extension loading enabled
2020-02-11 16:40:10: -- Loading ~/.hammerspoon/init.lua
2020-02-11 16:40:10: -- Loading extension: alert
2020-02-11 16:40:10: *** ERROR: ...app/Contents/Resources/extensions/hs/_coresetup/init.lua:597: module 'core' not found:
	no field package.preload['core']
	no file '/Users/erik/.hammerspoon/core.lua'
	no file '/Users/erik/.hammerspoon/core/init.lua'
	no file '/Users/erik/.hammerspoon/Spoons/core.spoon/init.lua'
	no file '/usr/local/share/lua/5.3/core.lua'
	no file '/usr/local/share/lua/5.3/core/init.lua'
	no file '/usr/local/lib/lua/5.3/core.lua'
	no file '/usr/local/lib/lua/5.3/core/init.lua'
	no file './core.lua'
	no file './core/init.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.lua'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core/init.lua'
	no file '/Users/erik/.hammerspoon/core.so'
	no file '/usr/local/lib/lua/5.3/core.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './core.so'
	no file '/Applications/Hammerspoon.app/Contents/Resources/extensions/core.so'
stack traceback:
	[C]: in function 'rawrequire'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:597: in function 'require'
	/Users/erik/.hammerspoon/init.lua:6: in main chunk
	[C]: in function 'xpcall'
	...app/Contents/Resources/extensions/hs/_coresetup/init.lua:648: in function 'hs._coresetup.setup'
	(...tail calls...)

```
